### DISCUSSION TITLE
is it possible to run coffee editor as a desktop application?

### DISCUSSION THREAD

**Discussion by shinesamjacob** [eclipse-emfcloud/emfcloud#159] [Coffee Editor] [answered]
is it possible to run the coffee editor as a desktop application?

**Comment by eneufeld** [comment, marked as answer]
Hi @shinesamjacob ,
you could run it as an electron app.
The coffee editor currently does not includes the electron app but you should be able to add it using any of the existing theia electron configurations as an example: https://github.com/eclipse-theia/theia/tree/master/examples/electron

**Reply by shinesamjacob** [reply]
when we run the coffee editor as a desktop app, do we need to start the server in a console like browser-app?
Is it possible to run the coffee editor as a standalone application?(like normal RCP product)

**Reply by eneufeld** [reply]
Hi @shinesamjacob ,
as the coffee editor is a browser application per se, you always operate with a browser frontend.
Using electron gives you a dedicated customizable browser that runs as your desktop app see [electron webpage](https://www.electronjs.org/). By doing so all backend services that need to be started will be started for you. The user start experience is as if you start a classic rcp application.

**Reply by eneufeld** [reply]
Two examples of popular electron desktop apps are spotify and VSCode.
